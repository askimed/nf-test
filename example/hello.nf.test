nextflow_pipeline {

  name "Test Hello World"
  script "nextflow-io/hello"

  test("hello world example should start 4 processes") {
    expect {
      with(workflow) {
        assert success
        assert trace.tasks().size() == 4
        assert "Ciao world!" in stdout
        assert "Bonjour world!" in stdout
        assert "Hello world!" in stdout
        assert "Hola world!" in stdout
      }
    }
  }

}
