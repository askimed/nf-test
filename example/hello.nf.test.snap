{
    "hello world example should start 4 processes": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    "WARN: Task runtime metrics are not reported when using macOS without a container engine",
                    "Hola world!",
                    "",
                    "Ciao world!",
                    "",
                    "Hello world!",
                    "",
                    "Bonjour world!",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 4,
                    "tasksSucceeded": 4
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2023-09-03T11:00:53.244626"
    }
}